npm/networkPolicyController.go

```diff
@@ -0,0 +1,491 @@
+		// Even though UninitNpmChains return error, isAzureNpmChainCreated sets up false
+		// #1 When deletion event is requeued to workqueue and deleteNetworkPolicy function is called again, it is safe.
+		// #2 when a new network policy is added, the default Azure NPM chain can install.
+		c.isAzureNpmChainCreated = false
+		if err = iptMgr.UninitNpmChains(); err != nil {
+			return fmt.Errorf("[deleteNetworkPolicy] Error: failed to uninitialize azure-npm chains with err: %s", err)
+		}
+	}
+
+	delete(c.npMgr.RawNpMap, cachedNetPolKey)
+	metrics.NumPolicies.Dec()
+	return nil
+}
+
+func (c *networkPolicyController) createCidrsRule(ingressOrEgress, policyName, ns string, ipsetEntries [][]string, ipsMgr *ipsm.IpsetManager) error {
+	spec := append([]string{util.IpsetNetHashFlag, util.IpsetMaxelemName, util.IpsetMaxelemNum})
+
+	for i, ipCidrSet := range ipsetEntries {
+		if ipCidrSet == nil || len(ipCidrSet) == 0 {
+			continue
+		}
+		setName := policyName + "-in-ns-" + ns + "-" + strconv.Itoa(i) + ingressOrEgress
+		log.Logf("Creating set: %v, hashedSet: %v", setName, util.GetHashedName(setName))
+		if err := ipsMgr.CreateSet(setName, spec); err != nil {
+			return fmt.Errorf("[createCidrsRule] Error: creating ipset %s with err: %v", ipCidrSet, err)
+		}
+		for _, ipCidrEntry := range util.DropEmptyFields(ipCidrSet) {
+			// Ipset doesn't allow 0.0.0.0/0 to be added. A general solution is split 0.0.0.0/1 in half which convert to
+			// 1.0.0.0/1 and 128.0.0.0/1
+			if ipCidrEntry == "0.0.0.0/0" {
+				splitEntry := [2]string{"1.0.0.0/1", "128.0.0.0/1"}
+				for _, entry := range splitEntry {
+					if err := ipsMgr.AddToSet(setName, entry, util.IpsetNetHashFlag, ""); err != nil {
+						return fmt.Errorf("[createCidrsRule] adding ip cidrs %s into ipset %s with err: %v", entry, ipCidrSet, err)
+					}
+				}
+			} else {
+				if err := ipsMgr.AddToSet(setName, ipCidrEntry, util.IpsetNetHashFlag, ""); err != nil {
+					return fmt.Errorf("[createCidrsRule] adding ip cidrs %s into ipset %s with err: %v", ipCidrEntry, ipCidrSet, err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *networkPolicyController) removeCidrsRule(ingressOrEgress, policyName, ns string, ipsetEntries [][]string, ipsMgr *ipsm.IpsetManager) error {
+	for i, ipCidrSet := range ipsetEntries {
+		if ipCidrSet == nil || len(ipCidrSet) == 0 {
+			continue
+		}
+		setName := policyName + "-in-ns-" + ns + "-" + strconv.Itoa(i) + ingressOrEgress
+		log.Logf("Delete set: %v, hashedSet: %v", setName, util.GetHashedName(setName))
+		if err := ipsMgr.DeleteSet(setName); err != nil {
+			return fmt.Errorf("[removeCidrsRule] deleting ipset %s with err: %v", ipCidrSet, err)
+		}
+	}
+
+	return nil
+}
+
+// GetProcessedNPKey will return netpolKey
+// (TODO): will use this function when optimizing management of multiple network policies with merging and deducting multiple network policies.
+func (c *networkPolicyController) getProcessedNPKey(netPolObj *networkingv1.NetworkPolicy) string {
```

Comment:
Can you add a testcase for this function.

Reply:
It would be better when we start using this function with network policy management (e.g., addpolicy and deductpolicy). I will comment it out to remove coverage.